app/controllers/recipes_controller.rb

```diff
@@ -9,10 +9,14 @@ def new
     @recipe = Recipe.new
   end
 
+  def show
+    @recipe = Recipe.find(params[:id])
+  end
+
   def create
-    @recipe = current_user.recipes.new(permitted_params)
+    recipe_generator = RecipeGeneratorService.new(permitted_params[:ingredients], current_user.id).call
```

Comment:
This is great ! 👏 

We could keep using `@recipe =` (instead of `recipe_generator`) here, because the service call will return a recipie object and that way we keep consistency with other methods in the controller.